src/hooks/useErc20.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react'
+
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+
+import { Network } from 'types'
+
+import { buildErc20Key, saveSingleErc20 } from 'reducers-actions/erc20'
+
+import useGlobalState from 'hooks/useGlobalState'
+
+import { getErc20Info } from 'services/helpers'
+
+import { web3, erc20Api } from 'apps/explorer/api'
+
+import { ExplorerAppState } from 'apps/explorer/state'
+
+type Params = {
+  address?: string
+  networkId?: Network
+}
+
+type Return = { isLoading: boolean; error?: string; value?: TokenErc20 }
+
+export function useErc20(params: Params): Return {
+  const { address, networkId } = params
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const [{ erc20s }, dispatch] = useGlobalState<ExplorerAppState>()
```

Comment:
I know we did it like this in the past, but why not using `useSelector` https://react-redux.js.org/api/hooks

Reply:
I was not aware of this. Looks fun!
Maybe talk about that tomorrow?
I have already way too many open waterfall PRs :sweat_smile: 
And this will probably clear out a bunch of code in those PRs... :shrug: